app/src/main/java/memphis/myapplication/UI/SendFriendRequestFragment.java

```diff
@@ -80,19 +89,16 @@ protected void onCreate(Bundle savedInstanceState) {
             }
             message.setText(friendsMessage);
         }
-        realm.close();
 
         // Radio buttons
-        RadioButton rdb1 = findViewById(R.id.radioMutualFriend);
+        RadioButton rdb1 = sendFriendRequestView.findViewById(R.id.radioMutualFriend);
         rdb1.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 // Iterate through our mutual friends
                 String friendsMessage = "";
-                Realm realm = Realm.getDefaultInstance();
-                User newFriend = realm.where(User.class).equalTo("username", friend).findFirst();
-                ArrayList<String> newFriendsList = newFriend.getFriends();
-                RealmResults<User> myFriends = realm.where(User.class).equalTo("trust", true).findAll();
+                ArrayList<String> newFriendsList = databaseViewModel.getFriendsofFriend(friend);
+                ArrayList<User> myFriends = databaseViewModel.getTrustedFriends();
```

Comment:
I'm not sure exactly where the problem lies here, but in my tests, even when I share a mutual friend with another user, I still get the message "No mutual friends." I checked the database objects and the mutual friend exists in both, so the problem probably lies here. If you don't have an easy way to test with a third user, I can look into this more.

Reply:
Umm, I am limited by the ability to make friends among my devices (because of some reasons I will try to solve). So, it would be great if you could look into it.